pkg/autodiscovery/providers/docker.go

```diff
@@ -120,12 +120,17 @@ func (d *DockerConfigProvider) IsUpToDate() (bool, error) {
 func parseDockerLabels(containers map[string]map[string]string) ([]integration.Config, error) {
 	var configs []integration.Config
 	for cID, labels := range containers {
-		c, errors := extractTemplatesFromMap(docker.ContainerIDToEntityName(cID), labels, dockerADLabelPrefix)
+		dockerEntityName := docker.ContainerIDToEntityName(cID)
+		c, errors := extractTemplatesFromMap(dockerEntityName, labels, dockerADLabelPrefix)
```

Comment:
I thought about it, but `extractTemplatesFromMap` is also called in some corner cases from (with the cluster agent I think).

This function seems more like a general than a provider, I chose to move the logic in the providers themselves to avoid any side effect.